core/src/main/java/io/fabric8/maven/core/util/MavenUtil.java

```diff
@@ -180,10 +184,17 @@ public static boolean hasClass(MavenProject project, String ... classNames) {
     }
 
     /**
-     * Returns the spring boot `application.properties` or an empty properties object if not found
+     * Returns the spring boot configuration (supports `application.properties` and `application.yml`)
+     * or an empty properties object if not found
      */
     public static Properties getSpringBootApplicationProperties(MavenProject project) {
-        return getPropertiesFile(project, "application.properties");
+        URLClassLoader compileClassLoader = getCompileClassLoader(project);
```

Comment:
Not your fault, but more a reminder to myself: This is the wrong class for spring-boot handling, In fact its even the wrong module. All specific tech stuff (spring-boot, wildfly-swarm, karaf, tomcat, ...) must be in generators and enrichers which nicely encapsulates these tech. See also #301 and comments to #503 .